urllib3/util/connection.py

```diff
@@ -92,7 +80,6 @@ def create_connection(address, timeout=socket._GLOBAL_DEFAULT_TIMEOUT,
             err = e
             if sock is not None:
                 sock.close()
-                sock = None
```

Comment:
sock gets set to None at the top of the loop regardless of if it's set at the bottom. I will restore it.